### PR TITLE
pallet-staking: Reorg migration, pallet and pallet impls

### PR DESCRIPTION
**Pull request by emostov**
closes https://github.com/paritytech/substrate/issues/9408

The main thing left out of this is moving the storage items and types to their own module (they are left in lib.rs for now). I think that should be easy to do in a follow up PR and leaving it out now will make it _slightly_ easier to merge this change into other open PRs that touch staking.

Note, the visibility on few functions in `impl.rs` was increased from either private or `pub(super)` to `pub(crate)`.

cc @kianenigma

### CODE REVIEW

**Comment by emostov** [comment]
@kianenigma I am thinking for now we can just move the migrations module. And then once the bags PR becomes very stable we can do a bigger re-org. My concern with doing a big re-org now is that transferring those changes to the bags PR would be difficult and may result in us losing some details without realizing.

**Comment by kianenigma** [comment]
I am fine with both. The bags PR has a small footpring in the rest of the pallet, so I think it is possible to do the full move as well. Nonetheless, this is not worth risking anything. We can always do this later.

File: frame/staking/src/migrations.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
+use super::*;
```
Comment (kianenigma) [inline frame/staking/src/migrations.rs:1]: license

File: frame/staking/src/lib.rs [hunk 2]
```diff
@@ -3415,6 +880,16 @@ where
 	}
 }
 
+/// A `Convert` implementation that finds the stash of the given controller account,
+/// if any.
+pub struct StashOf<T>(sp_std::marker::PhantomData<T>);
```
Comment (kianenigma) [inline frame/staking/src/lib.rs:885]: `types.rs`?

**Review by kianenigma — APPROVED** [review]
Add license header and an appropriate `//! single line of module doc` to each new file, and I am all fine with this change.

File: frame/staking/src/lib.rs (hunk 2 above)
Comment (emostov) [reply frame/staking/src/lib.rs:885]: Yeah I was thinking something like that for a follow up - we can  have `crate::pallet::types` and `crate::types` for ones we may want to expose publicly.

**Review by shawntabrizi — APPROVED** [review]
don't really see the need for the `pallet` folder, rather than a simple `lib.rs` file for that stuff, but okay.

**Comment by emostov** [comment]
bot merge

**Pull request merged**